src/clarity/datagrid/_datagrid.clarity.scss

```diff
@@ -104,10 +105,21 @@
                 }
             }
 
+            .datagrid-column-flex {
+                display: flex;
+            }
+
+            clr-dg-filter,
+            clr-dg-string-filter {
+                display: flex;
+                order: 99;
+                margin-left: auto;
+            }
+
             &.asc, &.desc {
                 font-weight: 600;
 
-                &::after {
+                .datagrid-column-flex::after {
```

Comment:
This is missing `flex: 0 0 auto` The arrow icon shouldn't flex.

Checked your branch right now, this is what it looks like at the moment:
![image](https://cloud.githubusercontent.com/assets/4647197/21157434/46c2e9c6-c147-11e6-837f-2acbe6be4ee8.png)
And this is with `flex: 0 0 auto`:
![image](https://cloud.githubusercontent.com/assets/4647197/21157448/56f77618-c147-11e6-8beb-58af60af634d.png)